### PR TITLE
feat: add account address to ga tracking

### PR DESCRIPTION
**Pull request by nenadV91**
# Summary

Adds functionality to track user address as a dimension and also as user id.
The weird value ``"${account}"`` is because in google analytics it will be parsed as a number if we just pass the `account`
Screenshot 2023-05-11 at 12 32 04

# To test
The way I tested this is by GA debug extension, there should be dimension4 in events once you connect your wallet
Screenshot 2023-05-11 at 12 49 56

And also in the GA console
Screenshot 2023-05-11 at 12 47 40
Screenshot 2023-05-11 at 12 47 54

### CODE REVIEW

**Comment by elena-zh** [comment]
Hey @nenadV91 , changes LGTM. However, 1 question: I see no events when I change accounts in MM. Is this OK? 
change account

**Comment by nenadV91** [comment]
@elena-zh It should be changed for next events from that point on.

**Review by elena-zh — APPROVED** [review]

File: src/components/analytics/hooks/useAnalyticsReporter.ts [hunk 1]
```diff
@@ -71,6 +72,10 @@ export function useAnalyticsReporter() {
     // custom dimension 2 - walletname
     googleAnalytics.setDimension(Dimensions.walletName, account ? walletName : 'Not connected')
 
+    // Set userId and also dimensions because ReactGA.set might not be working
+    googleAnalytics.setDimension(Dimensions.userAddress, account)
+    ReactGA.set({ userId: account })
```
Comment (alfetopito) [inline src/components/analytics/hooks/useAnalyticsReporter.ts:77]: Aren't you sure it works?
How can you test it?

**Review by alfetopito — COMMENTED** [review]
I haven't seen `"${account}"` that you mentioned in the description anywhere.
Where should I see it?

**Review by shoom3301 — APPROVED** [review]

File: src/components/analytics/hooks/useAnalyticsReporter.ts (hunk 1 above)
Comment (nenadV91) [reply src/components/analytics/hooks/useAnalyticsReporter.ts]: That seems to be code before the [second commit](https://github.com/cowprotocol/cowswap/pull/2450/commits/36f05c6f7df80e7e041b776a2793f84b0fa0d95f), now it looks like this 
Screenshot 2023-05-15 at 10 11 51

I've tested it in GA dashboard but if its not passed with quotes we get the first item like here
Screenshot 2023-05-15 at 10 14 43

Comment (alfetopito) [reply src/components/analytics/hooks/useAnalyticsReporter.ts]: In that case (if it doesn't work), we might not use `ReactGA.set` at all, no?

**Pull request merged**